### PR TITLE
Decode a Morse code message

### PR DESCRIPTION
**Pull request by lily-coder**
### **Decode a Morse code message**
In this project we:
Create a method to decode a Morse code character, take a string parameter, and return the corresponding character in uppercase

Create a method to decode an entire word in Morse code, take a string parameter, and return the string representation. Every character in a word will be separated by a single space.

Create a method to decode the entire message in Morse code, take a string parameter, and return the string representation. Every word will be separated by 3 spaces 

We have implemented all the changes required. This is our last request kindy check it out.

### CODE REVIEW

File: my_enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    puts key.to_s if char == value
+  end
+end
+
+decode_char(values)
+
+puts 'Please enter a group of characters'
+words = gets.chomp
+
+def decode_word(word)
```
Comment (SajjadAhmad14) [inline my_enumerable.rb]: #decode_word should reuse #decode_char.

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  end
+  puts actual_word.join
+end
+
+decode_word(words)
+
+puts 'Please enter a sentence'
+input_sentences = gets.chomp
+
+def decode_sentence(sentence)
```
Comment (SajjadAhmad14) [inline my_enumerable.rb]: #decode_sentence should reuse  #decode_word.

File: my_enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  sentences = []
```
Comment (SajjadAhmad14) [inline my_enumerable.rb]: #decode_sentence("-- -.--   -. .- -- .") should return "MY WORD" but it returns "MY NAME". Please correct the issues.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ashley and Lilian, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: decode_morse.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  end
+end
+
+def decode_word(str)
+  arr = []
+  str.split.each do |n|
+    arr.push(MORSE_CODE[n])
+  end
+  arr.join
+end
```
Comment (SajjadAhmad14) [inline decode_morse.rb]: Thank you for using the DRY rule for the `#decode` method.
`#decode_word` should also follow the DRY rule and reuse the `#decode_char` method to translate a single char symbol into a letter.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ashley and Lilian, 
 ### Thank you for working on the changes suggested previously.  :ok_hand:

## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: decode_morse.rb (hunk 4 above)
Comment (lily-coder) [reply decode_morse.rb:44]: Its not possible please

Comment (lily-coder) [reply decode_morse.rb]: Thank you.
We have implemented the change and now #decode_word is following the DRY rule and reusing the #decode_char method to translate a single char symbol into a letter.

**Review by ZeenatLawal — APPROVED** [review]
Hi @lily-coder and @Isabellanamutenda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 
6vEo


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by lily-coder** [comment]
Revert

**Pull request merged**